src/mozanalysis/frequentist_stats/linear_models/functions.py

```diff
@@ -355,66 +425,112 @@ def summarize_joint(
-        )
-
-        abs_uplifts = _extract_relative_uplifts(
+        abs_uplifts = _extract_absolute_uplifts(
             results, branch, ref_branch_label, alphas
         )
 
+        if (
+            covariate_col_label is None
+            or covariate_col_label not in results.params.index
+        ):
+            rel_uplifts = _extract_relative_uplifts(
+                results, branch, ref_branch_label, alphas, treatment_branches
+            )
+        else:
+            rel_uplifts = _extract_relative_uplifts(
+                results,
+                branch,
+                ref_branch_label,
+                alphas,
+                treatment_branches,
+                covariate_col_label,
+            )
         output[branch] = pd.concat([rel_uplifts, abs_uplifts])
 
+    if deallocate_aggressively:
+        import gc
+
+        del results
+        gc.collect()
+
     return output
 
 
-def make_model_df(
+def prepare_df_for_modeling(
     df: pd.DataFrame,
-    col_label: str,
-    covariate_col_label: str | None = None,
+    target_col: str,
     threshold_quantile: float | None = None,
+    covariate_col: str | None = None,
+    copy: bool = True,
 ) -> pd.DataFrame:
-    """Prepares a dataset for modeling. Removes nulls from the response variable
-    (col_label). Optionally, adds a similarly cleaned covariate column.
-    Optionally, applies outlier filtering (to both the response variable and
-    the covariate, if one exists).
+    """
+    Performs outlier clipping inplace and returns a view into the dataframe that can be
+    used for modeling: target and covariate, if passed, are guaranteed to be non-null.
 
     Parameters:
-    - df (pd.DataFrame): the raw, client-level data. Must have a column "branch".
-    - col_label (str): the label of the response variable. Must be a column in df. Only
-    non-null values are modeled.
-    - covariate_col_label (Optional[str]): the name of a covariate column to include.
-    Must be a column in df. If included, only non-null values are modeled.
-    - threshold_quantile (Optional[float]): the outlier threshold. See
-    `filter_outliers`.
-
-    Returns:
-    - model_df (pd.DataFrame): the cleaned data, ready for modeling.
+    - df (pd.DataFrame): a cleaned set of data, ready for modeling (such as the output
+    of `make_model_df`)
+    - target_col (str): the target variable for which inferences are desired.
+    - threshold_quantile (Optional[float]): the outlier threshold. See `filter_outliers`
+    - covariate_col_label (Optional[str]): the name of a covariate to include in the
+    model.
+    - copy (bool): if True (default) returns a cleaned copy of the data. If False,
+    modifies data inplace
     """
+    indexer = ~df[target_col].isna()
+    if covariate_col is not None:
+        indexer &= ~df[covariate_col].isna()
 
-    indexer = ~df[col_label].isna()
-    if covariate_col_label is not None:
-        indexer &= ~df[covariate_col_label].isna()
+    if copy:
```

Comment:
The intention was to be defensive, in the case where a data scientist was using the functions interactively. I wanted to avoid mutating their data out from under them.